hdr.go

```diff
@@ -331,21 +331,37 @@ func (h *Histogram) ValueAtPercentile(percentile float64) int64 {
 		percentile = 100
 	}
 
-	total := int64(0)
 	countAtPercentile := int64(((percentile / 100) * float64(h.totalCount)) + 0.5)
+	valueFromIdx := h.getValueFromIdxUpToCount(countAtPercentile)
+	if percentile == 0.0 {
+		return h.lowestEquivalentValue(valueFromIdx)
+	}
+	return h.highestEquivalentValue(valueFromIdx)
+}
 
-	i := h.iterator()
-	for i.nextCountAtIdx() {
-		total += i.countAtIdx
-		if total >= countAtPercentile {
-			if percentile == 0.0 {
-				return h.lowestEquivalentValue(i.valueFromIdx)
-			}
-			return h.highestEquivalentValue(i.valueFromIdx)
+func (h *Histogram) getValueFromIdxUpToCount(countAtPercentile int64) int64 {
+	var countToIdx int64 = 0
+	var valueFromIdx int64 = 0
+	var subBucketIdx int32 = -1
+	var bucketIdx int32 = 0
```

Comment:
it is normally not necessary to set var variable explicitly to 0

Reply:
Fixed on the latest commit